src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -60,12 +64,23 @@ public sealed partial class TarWriter : IDisposable
 
             entry._header._mode = (status.Mode & 4095); // First 12 bits
 
-            entry.Uid = (int)status.Uid;
-            entry.Gid = (int)status.Gid;
+            // Uid and UName
+            entry._header._uid = (int)status.Uid;
+            if (!_userIdentifiers.TryGetValue(status.Uid, out string? uName))
+            {
+                uName = Interop.Sys.GetUserNameFromPasswd();
```

Comment:
Don't you want to pass in `status.Uid` to this method?

Reply:
And if so, and all the Tar tests are passing, then I think we have a hole in our tests here.

Reply:
Right. Fixed.